desktop/app/window-handlers/external-links/index.js

```diff
@@ -7,24 +8,69 @@ let targetURL = '';
 module.exports = function ( { view } ) {
 	// TODO: Replace the "new-window" event with webcontents.setWindowOpenHandler
 	// when Electron is updated to >= 13.x
-	view.webContents.on( 'new-window', function ( event, url ) {
-		// Check if the incoming URL is blank and if it is send to the targetURL instead
-		const urlToLoad = url.includes( 'about:blank' ) || url === '' ? targetURL : url;
-		log.info( `Navigating to URL: '${ urlToLoad }'` );
+	view.webContents.on(
+		'new-window',
+		function (
+			event,
+			url,
+			frameName,
+			disposition,
+			options,
+			additionalFeatures,
+			referrer,
+			postBody
+		) {
+			if ( url.includes( 'https://accounts.google.com' ) ) {
+				event.preventDefault();
+				const win = new BrowserWindow( {
+					webContents: options.webContents, // use existing webContents if provided
+					show: false,
+				} );
 
-		event.preventDefault();
-		view.webContents.loadURL( urlToLoad );
-		return;
-	} );
+				win.webContents.on( 'will-navigate', function ( e, u ) {
+					log.info( 'Google Window will navigate: ' + u );
+				} );
+
+				win.webContents.on( 'will-redirect', function ( e, u ) {
+					log.info( 'Google Window will redirect: ' + u );
+				} );
+
+				win.once( 'ready-to-show', () => win.show() );
+				if ( ! options.webContents ) {
+					const loadOptions = {
+						httpReferrer: referrer,
+					};
+					if ( postBody != null ) {
+						const { data, contentType, boundary } = postBody;
+						loadOptions.postData = postBody.data;
+						loadOptions.extraHeaders = `content-type: ${ contentType }; boundary=${ boundary }`;
+					}
+
+					win.loadURL( url, loadOptions ); // existing webContents will be navigated automatically
+				}
+				event.newGuest = win;
+			} else {
+				// Check if the incoming URL is blank and if it is send to the targetURL instead
+				const urlToLoad = url.includes( 'about:blank' ) || url === '' ? targetURL : url;
+				log.info( `Navigating to URL: '${ urlToLoad }'` );
+
+				event.preventDefault();
+				view.webContents.loadURL( urlToLoad );
+				return;
+			}
+		}
+	);
 
 	// Magic links aren't supported in the app currently. Instead we'll show a message about how
 	// to set a password on the account to log in that way.
 	view.webContents.on( 'will-navigate', function ( event, url ) {
```

Comment:
We should get this fix into Calypso `trunk` as a separate patch, as it seems like it could affect other parts of the app.

Reply:
Good call, I've done that in this PR https://github.com/Automattic/wp-calypso/pull/55571